clients/axios/axios-client-sdk.spec.ts

```diff
@@ -0,0 +1,105 @@
+import { createUser } from "./sdk/client";
+import * as moxios from "moxios";
+import { CreateUserResponse } from "./sdk/types";
+
+const expectedResponse: CreateUserResponse = {
+  success: true,
+  created_at: "2018-01-01"
+};
+
+describe("TypeScript axios client sdk test", () => {
+  beforeEach(() => {
+    // import and pass your custom axios instance to this method
+    moxios.install();
+  });
+
+  afterEach(() => {
+    // import and pass your custom axios instance to this method
+    moxios.uninstall();
+  });
+
+  describe("POST request", () => {
+    it("pass correct request body", async () => {
+      moxios.stubRequest("/users/create", {
+        status: 200,
+        response: expectedResponse
+      });
+
+      await createUser({ name: "User 1", roles: "admin" }, "test-token");
+
+      const request = moxios.requests.mostRecent();
+      expect(request.config.method).toBe("post");
+      expect(JSON.parse(request.config.data)).toStrictEqual({
+        name: "User 1",
+        roles: "admin"
+      });
+    });
+
+    it("pass correct header", async () => {
+      moxios.stubRequest("/users/create", {
+        status: 200,
+        response: expectedResponse
+      });
+
+      await createUser({ name: "User 1", roles: "admin" }, "test-token");
+
+      const request = moxios.requests.mostRecent();
+      expect(request.config.method).toBe("post");
+      expect(request.config.headers["Authorization"]).toBe("test-token");
+    });
+
+    it("allows undefined header", async () => {
+      moxios.stubRequest("/users/create", {
+        status: 200,
+        response: expectedResponse
+      });
+
+      await createUser({ name: "User 1", roles: "admin" }, undefined);
+      const request = moxios.requests.mostRecent();
+      expect(request.config.method).toBe("post");
+      expect(request.config.headers["Authorization"]).toBeUndefined();
+    });
+
+    it("can handle successful request", async () => {
+      moxios.stubRequest("/users/create", {
+        status: 200,
+        response: expectedResponse
+      });
+
+      const response = await createUser(
+        { name: "User 1", roles: "admin" },
+        "test-token"
+      );
+      expect(response.kind).toBe("success");
+      expect(response.data).toEqual(expectedResponse);
+    });
+
+    it("throws error when response is not CreateUserResponse", async () => {
```

Comment:
I'd love to see another PR that adds thorough tests for the validators themselves.